### PR TITLE
Feat : Add color to custom navigation items

### PR DESCRIPTION
**Pull request by ibrahimBougaoua**
I searched in the docs and did not find how to change the color of navigation items like User Menu Items, so I added the color to navigation items.

- [ ] Changes have been thoroughly tested to not break existing functionality.
- [ ] New functionality has been documented or existing documentation has been updated to reflect changes.
- [ ] Visual changes are explained in the PR description using a screenshot/recording of before and after.

### CODE REVIEW

**Review by sporho-3665 — COMMENTED** [review]
thank you

**Comment by danharrin** [comment]
Please send screenshots as the PR template mentions

**Comment by ibrahimBougaoua** [comment]
> Please send screenshots as the PR template mentions

**I added the color to custom navigation items.**

Filament::registerNavigationItems([
    NavigationItem::make('Analytics 1')
        ->url('https://filament.pirsch.io', shouldOpenInNewTab: true)
        ->icon('heroicon-o-presentation-chart-line')
        **->iconColor('danger')**
        ->activeIcon('heroicon-s-presentation-chart-line')
        ->group('Reports')
        ->sort(3),
    NavigationItem::make('Analytics 2')
        ->url('https://filament.pirsch.io', shouldOpenInNewTab: true)
        ->icon('heroicon-o-presentation-chart-line')
        **->iconColor('danger')**
        ->activeIcon('heroicon-s-presentation-chart-line')
        ->group('Reports')
        ->sort(3),
    NavigationItem::make('Analytics 3')
        ->url('https://filament.pirsch.io', shouldOpenInNewTab: true)
        ->icon('heroicon-o-presentation-chart-line')
        **->iconColor('success')**
        ->activeIcon('heroicon-s-presentation-chart-line')
        ->group('Reports')
        ->sort(3),
    NavigationItem::make('Analytics 4')
        ->url('https://filament.pirsch.io', shouldOpenInNewTab: true)
        ->icon('heroicon-o-presentation-chart-line')
        **->iconColor('primary')**
        ->activeIcon('heroicon-s-presentation-chart-line')
        ->group('Reports')
        ->sort(3),
]);

captue

**Comment by zepfietje** [comment]
As this only affects icons, could you update the API to `iconColor()` instead?
Mark as ready when done, thanks!

**Comment by ibrahimBougaoua** [comment]
> As this only affects icons, could you update the API to `iconColor()` instead? Mark as ready when done, thanks!

**it's done,thanks.**

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php [hunk 1]
```diff
@@ -8,12 +9,18 @@
...
 ])
 
-<li
-    @class([
-        'filament-sidebar-item overflow-hidden',
-        'filament-sidebar-item-active' => $active,
-    ])
->
+@php
+    $iconClasses = \Illuminate\Support\Arr::toCssClasses([
```
Comment (zepfietje) [inline packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php]: No need to extract if only used in 1 place.

File: packages/admin/src/Navigation/NavigationItem.php [hunk 2]
```diff
@@ -10,6 +10,8 @@ class NavigationItem
 
     protected ?Closure $isActiveWhen = null;
 
+    protected ?string $iconColor = null;
```
Comment (zepfietje) [inline packages/admin/src/Navigation/NavigationItem.php]: Please move after `$icon`.

File: packages/admin/src/Navigation/NavigationItem.php [hunk 3]
```diff
@@ -53,6 +55,13 @@ public function group(?string $group): static
         return $this;
     }
 
+    public function iconColor(?string $iconColor): static
```
Comment (zepfietje) [inline packages/admin/src/Navigation/NavigationItem.php]: Move this one too.

File: packages/admin/src/Navigation/NavigationItem.php [hunk 4]
```diff
@@ -118,6 +127,11 @@ public function getGroup(): ?string
         return $this->group;
     }
 
+    public function getIconColor(): ?string
```
Comment (zepfietje) [inline packages/admin/src/Navigation/NavigationItem.php]: And this one as well.
Should come after icon.

**Review by zepfietje — CHANGES_REQUESTED** [review]

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php (hunk 1 above)
Comment (ibrahimBougaoua) [reply packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php]: i don't understand.

@php
    $iconClasses = \Illuminate\Support\Arr::toCssClasses([
        'h-5 w-5 shrink-0',
        'text-primary-500' => $iconColor === 'primary',
        'text-danger-500' => $iconColor === 'danger',
        'text-gray-500' => $iconColor === 'secondary',
        'text-success-500' => $iconColor === 'success',
        'text-warning-500' => $iconColor === 'warning',
    ]);
@endphp

Comment (zepfietje) [reply packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php]: There's no need to extract the classes into an `$iconClasses` variable if that variable is only used in a single place in the template. Or am I missing something?

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php [hunk 5]
```diff
@@ -8,12 +9,18 @@
...
+        'h-5 w-5 shrink-0',
+        'text-primary-500' => $iconColor === 'primary',
+        'text-danger-500' => $iconColor === 'danger',
+        'text-gray-500' => $iconColor === 'secondary',
+        'text-success-500' => $iconColor === 'success',
+        'text-warning-500' => $iconColor === 'warning',
+    ]);
+@endphp
+
+<li @class(['filament-sidebar-item overflow-hidden', 'filament-sidebar-item-active' => $active])>
```
Comment (zepfietje) [inline packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php]: Please undo this formatting change.

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php (hunk 1 above)
Comment (ibrahimBougaoua) [reply packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php]: **iconColor** have 5 possible values [ primary,danger,secondary,success or primary ] , so without the code provided how i can select the color ?

in the case where i pass the color class directly like "text-primary-500" sure i dont need the code i provided above .

NavigationItem::make('Analytics 1')
                    **->iconColor('danger')** // **_i pass "danger" not text-danger-500 , so without this code above how i can select the correct color that match with the parameter i passed "danger"._** 

and if i'm missing something could  please u tell me some other ways to implement the solution i nedded following the best practice and leaving every clean and clear .

Comment (ibrahimBougaoua) [reply packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php:13]: > **iconColor** have 5 possible values [ primary,danger,secondary,success or primary ] , so without the code provided how i can select the color ?
> 
> in the case where i pass the color class directly like "text-primary-500" sure i dont need the code i provided above .
> 
> NavigationItem::make('Analytics 1') **->iconColor('danger')** // **_i pass "danger" not text-danger-500 , so without this code above how i can select the correct color that match with the parameter i passed "danger"._**

and if i'm missing something please could u tell me some other way to implement the solution i nedded in the best way that exist

Comment (zepfietje) [reply packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php]: The API doesn't need changing. I'm just saying storing it in the $iconClasses variable in the view isn't needed as the classes can be applied to the HTML element using the `@class()` Blade directive directly. 🙂

Comment (ibrahimBougaoua) [reply packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php]: it's done,thanks.

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php [hunk 6]
```diff
@@ -1,4 +1,5 @@
 @props([
+    'iconColor' => null,
```
Comment (zepfietje) [inline packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php]: Could you move this after `icon`?

File: packages/admin/src/Navigation/NavigationItem.php [hunk 7]
```diff
@@ -117,12 +125,17 @@ public function getGroup(): ?string
     {
         return $this->group;
     }
-
+
```
Comment (zepfietje) [inline packages/admin/src/Navigation/NavigationItem.php]:
```suggestion

```

File: packages/admin/src/Navigation/NavigationItem.php [hunk 8]
```diff
@@ -117,12 +125,16 @@ public function getGroup(): ?string
     {
         return $this->group;
     }
-
     public function getIcon(): string
```
Comment (zepfietje) [inline packages/admin/src/Navigation/NavigationItem.php]:
```suggestion

    public function getIcon(): string
```

File: packages/admin/src/Navigation/NavigationItem.php (hunk 4 above)
Comment (ibrahimBougaoua) [reply packages/admin/src/Navigation/NavigationItem.php]: it's done,thanks.

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php (hunk 6 above)
Comment (ibrahimBougaoua) [reply packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php]: it's done,thanks.

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php [hunk 9]
```diff
@@ -39,7 +45,14 @@
...
         <x-dynamic-component
             :component="($active && $activeIcon) ? $activeIcon : $icon"
-            class="h-5 w-5 shrink-0"
+            @class([
+                'h-5 w-5 shrink-0',
+                'text-primary-500' => $color === 'primary',
+                'text-danger-500' => $color === 'danger',
+                'text-gray-500' => $color === 'secondary',
+                'text-success-500' => $color === 'success',
+                'text-warning-500' => $color === 'warning',
```
Comment (danharrin) [inline packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php]:
```suggestion
                'text-primary-500' => $iconColor === 'primary',
                'text-danger-500' => $iconColor === 'danger',
                'text-gray-500' => $iconColor === 'secondary',
                'text-success-500' => $iconColor === 'success',
                'text-warning-500' => $iconColor === 'warning',
```

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php [hunk 10]
```diff
@@ -39,7 +45,14 @@
...
         <x-dynamic-component
             :component="($active && $activeIcon) ? $activeIcon : $icon"
-            class="h-5 w-5 shrink-0"
+            @class([
+                'h-5 w-5 shrink-0',
+                'text-primary-500' => $iconColor === 'primary',
+                'text-danger-500' => $iconColor === 'danger',
+                'text-gray-500' => $iconColor === 'secondary',
+                'text-success-500' => $iconColor === 'success',
+                'text-warning-500' => $iconColor === 'warning',
```
Comment (danharrin) [inline packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php]:
```suggestion
                'text-primary-500' => (! $active) && ($iconColor === 'primary'),
                'text-danger-500' => (! $active) && ($iconColor === 'danger'),
                'text-gray-500' => (! $active) && ($iconColor === 'secondary'),
                'text-success-500' => (! $active) && ($iconColor === 'success'),
                'text-warning-500' => (! $active) && ($iconColor === 'warning'),
```

**Comment by danharrin** [comment]
Thanks @ibrahimBougaoua

**Pull request merged**